sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/ScenarioTests/DnsResolverTests.cs

```diff
@@ -0,0 +1,643 @@
+            updateDnsResolverAction.Should().Throw<CloudException>().Which.Response.StatusCode.Should().Be(HttpStatusCode.PreconditionFailed);
+        }
+
+        [Fact]
+        public void PutDnsResolver_ResolverExistsWithoutTagsToResolverWithTags_ExpectSuccess()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var dnsResolverName = TestDataGenerator.GenerateDnsResolverName();
+            var dnsResolver = TestDataGenerator.GenerateDnsResolver(
+                location: Constants.DnsResolverLocation,
+                subscriptionId: this.SubscriptionId,
+                resourceGroupName: resourceGroupName);
+            var createdDnsResolver = this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                 resourceGroupName: resourceGroupName,
+                 dnsResolverName: dnsResolverName,
+                 parameters: dnsResolver);
+            var addedTags = TestDataGenerator.GenerateTags();
+            dnsResolver.Tags = addedTags;
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: dnsResolver);
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.Tags.Should().NotBeNull().And.BeEquivalentTo(addedTags);
+        }
+
+        [Fact]
+        public void PutDnsResolver_ResolverExistsWithTagsToResolverWithTags_ExpectSuccess()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var dnsResolverName = TestDataGenerator.GenerateDnsResolverName();
+            var dnsResolver = TestDataGenerator.GenerateDnsResolver(
+                location: Constants.DnsResolverLocation, 
+                subscriptionId: this.SubscriptionId, 
+                resourceGroupName: resourceGroupName, 
+                tags: TestDataGenerator.GenerateTags());
+            var createdDnsResolver = this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                 resourceGroupName: resourceGroupName,
+                 dnsResolverName: dnsResolverName,
+                 parameters: dnsResolver);
+            var updatedDnsResolverTags = TestDataGenerator.GenerateTags(startFrom: dnsResolver.Tags.Count);
+            dnsResolver.Tags = updatedDnsResolverTags;
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: dnsResolver);
+
+            updatedDnsResolver.Should().NotBeNull();
+            updatedDnsResolver.Tags.Should().NotBeNull().And.BeEquivalentTo(updatedDnsResolverTags);
+        }
+
+
+        [Fact]
+        public void PutDnsResolver_ResolverExists_ExpectConflicts()
```

Comment:
I don't understand this case. Resolver exists and calling Put on it throws Conflict? That doesn't seem right.

Reply:
This case, the dns resolver name is used by that VNet A, then we try to upsert same dns resolver name with VNet B.

Reply:
```

        [Fact]
        public void PutDnsResolver_ResolverExistsUpdateVirtualNetwork_ExpectConflicts()
        {
            var resourceGroupName = this.CreateResourceGroup().Name;
            var existingDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName);
            var dnsResolverForUpdate = GenerateDnsResolverWithNewlyCreatedVirtualNetwork(location: Constants.DnsResolverLocation, resourceGroupName: resourceGroupName);

            Action updateDnsResolverAction = () => this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
                resourceGroupName: resourceGroupName,
                dnsResolverName: existingDnsResolver.Name,
                parameters: dnsResolverForUpdate);

            updateDnsResolverAction.Should().NotBeNull();
            updateDnsResolverAction.Should().Throw<CloudException>().Which.Response.StatusCode.Should().Be(HttpStatusCode.Conflict);
        }
```

Reply:
Updated as above, hope that make the test case less confusing.